***Consider a box of length equal to $l=0.8$ units, breadth equal to $w=0.4$ units and height equal to $h=0.2$ units.  Consider an orthonormal frame $\mathbf{e}$ with origin coinciding with one of the bottom vertices and the axis aligning along the edges of the box.***




Let $R\in SO(3)$ be a rotation that is composed of a rotation of $\theta_1=\pi/3$ rads about the  $1^{\mathrm{st}}$ axis and then by a rotation of $\theta_3=\pi/6$ rads about the $3^{\mathrm{rd}}$ axis.

Let $P_i$ denote the $3\times 1$ column representation of the $i^{\mathrm{th}}$ vertex for $i=1,2,\cdots, 8$


Answer the following:

* Plot the frame and vertices of the box in the same figure and attach the figure in the space below

* Consider the set of points that have the representation $RP_1,RP_2,\cdots,RP_8$ and plot them in the same figure that was generated above (using a different color for the points) and attach the figure in the space below

In [ ]:
from sympy import *
import plotly.graph_objects as go
import numpy as np
init_printing()
x, y, z = symbols('x y z')
theta, phi = symbols('theta phi')

In [ ]:
Fig1=go.Figure(
    data=[go.Scatter3d(x=[0,0.8,0.8,0,0,0.8,0,0.8],y=[0,0,0.4,0.4,0,0,0.4,0.4],z=[0,0,0,0,0.2,0.2,0.2,0.2],name="Vertices",mode="markers")],
    layout=go.Layout(
        scene=dict(
        xaxis=dict(range=[-1,1],autorange=False),
        yaxis=dict(range=[-1,1],autorange=False),
        zaxis=dict(range=[-0.5,0.5],autorange=False),
        aspectratio=dict(x=1, y=1, z=1))
    )
)
Fig1.add_trace(go.Scatter3d(x=[0,1],y=[0,0],z=[0,0],name="Xaxis",mode="lines",line=dict(color="red",width=10)))
Fig1.add_trace(go.Scatter3d(x=[0,0],y=[0,1],z=[0,0],name="Yaxis",mode="lines",line=dict(color="red",width=10)))
Fig1.add_trace(go.Scatter3d(x=[0,0],y=[0,0],z=[0,1],name="Zaxis",mode="lines",line=dict(color="red",width=10)))


Fig1.show()

So the rotational matrix,

\begin{align}
R_1(\frac{\pi}{3})&=  \begin{bmatrix}
  1&0&0\\
  0&cos\frac{\pi}{3}&sin\frac{\pi}{3}\\
  0&-sin\frac{\pi}{3}&cos\frac{\pi}{3}
  \end{bmatrix}\\
&=\begin{bmatrix}
  1&0&0\\
  0&\frac{1}{2}&\frac{\sqrt{3}}{2}\\
  0&-\frac{\sqrt{3}}{2}&\frac{1}{2}
  \end{bmatrix}
\end{align}

$%$

\begin{align}
R_2(0)&=  \begin{bmatrix}
  cos0&0&sin0\\
  0&1&0\\
  -sin0&0&cos0
  \end{bmatrix}\\
  &=  \begin{bmatrix}
  1&0&0\\
  0&1&0\\
  0&0&1
  \end{bmatrix}
\end{align}

$%$

\begin{align}
R_3(\frac{\pi}{6})&=  \begin{bmatrix}
  cos\frac{\pi}{6}&-sin\frac{\pi}{6}&0\\
  sin\frac{\pi}{6}&-cos\frac{\pi}{6}&0\\
  0&0&1
  \end{bmatrix}\\
  &=\begin{bmatrix}
  \frac{\sqrt{3}}{2}&-\frac{1}{2}&0\\
  \frac{1}{2}&-\frac{\sqrt{3}}{2}&0\\
  0&0&1
  \end{bmatrix}
\end{align}

In [ ]:
import math

Rt1=Matrix([[1,0,0],[0,1/2,math.sqrt(3)/2],[0,-math.sqrt(3)/2,1/2]])

Rt3=Matrix([[math.sqrt(3)/2,-1/2,0],[1/2,-math.sqrt(3)/2,0],[0,0,1]])

Rt=Rt1*Rt3
print(latex(Rt))

\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]


In [ ]:
x=[0,0.8,0.8,0,0,0.8,0,0.8]
y=[0,0,0.4,0.4,0,0,0.4,0.4]
z=[0,0,0,0,0.2,0.2,0.2,0.2]

for i in range(8):
  P=Matrix([[x[i],y[i],z[i]]])
  Q=P*Rt

  print(latex(Q))






\left[\begin{matrix}0 & 0 & 0\end{matrix}\right]
\left[\begin{matrix}0.692820323027551 & -0.4 & 0\end{matrix}\right]
\left[\begin{matrix}0.792820323027551 & -0.573205080756888 & 0.346410161513775\end{matrix}\right]
\left[\begin{matrix}0.1 & -0.173205080756888 & 0.346410161513775\end{matrix}\right]
\left[\begin{matrix}-0.0866025403784439 & 0.15 & 0.1\end{matrix}\right]
\left[\begin{matrix}0.606217782649107 & -0.25 & 0.1\end{matrix}\right]
\left[\begin{matrix}0.0133974596215561 & -0.0232050807568877 & 0.446410161513775\end{matrix}\right]
\left[\begin{matrix}0.706217782649107 & -0.423205080756888 & 0.446410161513775\end{matrix}\right]


\begin{align}
RP_1&=\begin{bmatrix}
0&0&0
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}0 & 0 & 0\end{matrix}\right]
\end{align}

$%$

\begin{align}
RP_2&=\begin{bmatrix}
0.8&0&0
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}0.692820323027551 & -0.4 & 0\end{matrix}\right]
\end{align}

$%$

\begin{align}
RP_3&=\begin{bmatrix}
0.8&0.4&0
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}0.792820323027551 & -0.573205080756888 & 0.346410161513775\end{matrix}\right]
\end{align}

$%$

\begin{align}
RP_4&=\begin{bmatrix}
0&0.4&0
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}0.1 & -0.173205080756888 & 0.346410161513775\end{matrix}\right]
\end{align}

$%$

\begin{align}
RP_5&=\begin{bmatrix}
0&0&0.2
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}-0.0866025403784439 & 0.15 & 0.1\end{matrix}\right]
\end{align}

$%$

\begin{align}
RP_6&=\begin{bmatrix}
0.8&0&0.2
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}0.606217782649107 & -0.25 & 0.1\end{matrix}\right]
\end{align}

$%$

\begin{align}
RP_7&=\begin{bmatrix}
0&0.4&0.2
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}0.0133974596215561 & -0.0232050807568877 & 0.446410161513775\end{matrix}\right]
\end{align}

$%$

\begin{align}
RP_8&=\begin{bmatrix}
0.8&0.4&0.2
\end{bmatrix}
\left[\begin{matrix}0.866025403784439 & -0.5 & 0\\0.25 & -0.433012701892219 & 0.866025403784439\\-0.433012701892219 & 0.75 & 0.5\end{matrix}\right]\\
&=\left[\begin{matrix}0.706217782649107 & -0.423205080756888 & 0.446410161513775\end{matrix}\right]]
\end{align}

In [ ]:
xx=[]
yy=[]
zz=[]

for m in range(8):
  P=Matrix([[x[m],y[m],z[m]]])
  Q=P*Rt
  xx.append(Q[0,0])
  yy.append(Q[0,1])
  zz.append(Q[0,2])

print(xx)
print(yy)
print(zz)

[0, 0.692820323027551, 0.792820323027551, 0.100000000000000, -0.0866025403784439, 0.606217782649107, 0.0133974596215561, 0.706217782649107]
[0, -0.400000000000000, -0.573205080756888, -0.173205080756888, 0.150000000000000, -0.250000000000000, -0.0232050807568877, -0.423205080756888]
[0, 0, 0.346410161513775, 0.346410161513775, 0.100000000000000, 0.100000000000000, 0.446410161513775, 0.446410161513775]


In [ ]:
Fig1.add_trace(go.Scatter3d(x=[0, 0.692820323027551, 0.792820323027551, 0.100000000000000, -0.0866025403784439, 0.606217782649107, 0.0133974596215561, 0.706217782649107],y=[0, -0.400000000000000, -0.573205080756888, -0.173205080756888, 0.150000000000000, -0.250000000000000, -0.0232050807568877, -0.423205080756888],z=[0, 0, 0.346410161513775, 0.346410161513775, 0.100000000000000, 0.100000000000000, 0.446410161513775, 0.446410161513775],name="RP",mode="markers",marker=dict(color="black",size = 8)))

Fig1.show()